### PR TITLE
Allow removing single patterns from default selection

### PR DESCRIPTION
**Pull request by asdil12**
eg. `PATTERNS=default,-gnome,web`

Needed for https://progress.opensuse.org/issues/33985

Verification: http://artemis.suse.de/tests/430#step/select_patterns_and_packages/75

### CODE REVIEW

File: tests/installation/select_patterns_and_packages.pm [hunk 1]
```diff
@@ -185,11 +186,19 @@ sub run {
             die "looping for too long" unless ($counter--);
             my $needs_to_be_selected;
             my $ret = check_screen('on-pattern', 1);
+            my $current_pattern = 'UNKNOWN_PATTERN';
 
             if ($ret) {    # unneedled pattern
                 for my $wp (keys %wanted_patterns) {
```
Comment (rwx788) [inline tests/installation/select_patterns_and_packages.pm]: I guess we need to rename wanted_patterns to something else, as now we have unwanted patterns there as well. But I like the idea!

Comment (asdil12) [reply tests/installation/select_patterns_and_packages.pm]: Renamed to `%patterns`

File: tests/installation/select_patterns_and_packages.pm [hunk 2]
```diff
@@ -175,25 +176,34 @@ sub run {
...
-            $wanted_patterns{$p} = 1;
+            $patterns{$p} = 1;
         }
 
         my $counter = 80;
         while (1) {
             die "looping for too long" unless ($counter--);
             my $needs_to_be_selected;
             my $ret = check_screen('on-pattern', 1);
+            my $current_pattern = 'UNKNOWN_PATTERN';
```
Comment (asmorodskyi) [inline tests/installation/select_patterns_and_packages.pm]: not sure that I understand purpose of this variable. Looks like you use it to detect if PATTERNS variable has at least one pattern to select and it is not default, right ?

File: tests/installation/select_patterns_and_packages.pm [hunk 3]
```diff
@@ -175,25 +176,36 @@ sub run {
...
+            $patterns{$p} = 1;
         }
 
         my $counter = 80;
         while (1) {
             die "looping for too long" unless ($counter--);
             my $needs_to_be_selected;
             my $ret = check_screen('on-pattern', 1);
+            # this variable will only be updated when the pattern list entry is under the cursor
+            # and the pattern is in the PATTERNS variable and there is a needle for that entry
```
Comment (asmorodskyi) [inline tests/installation/select_patterns_and_packages.pm]: thanks :)

**Pull request merged**